### PR TITLE
Schema API versioning

### PR DESCRIPTION
**Pull request by ddohler**
This sets up the RecordSchema viewset to generate versioned schemas. It also adds a "next_version" field to each RecordSchema which provides the next version of that schema, so that a set of versions is essentially a linked list.

This doesn't really deal much with ItemSchemas, except to keep their features roughly in line with RecordSchema. After discussing with @CloudNiner , we agreed that we're not totally clear on whether or how we'll be using ItemSchemas, so it seemed to make more sense to let them sit until we have a better idea of the use cases.

Another item that came up as a result of this is that it would probably be a good idea to represent Record Types as a separate model and reference them from RecordSchemas using a foreign key. This adds a to-do in the RecordSchema class.

@CloudNiner may want to take a look too.

### CODE REVIEW

File: app/ashlar/models.py [hunk 1]
```diff
@@ -25,8 +25,9 @@ class Meta(object):
 
 
 class SchemaModel(AshlarModel):
-    version = models.IntegerField()
+    version = models.PositiveIntegerField()
     schema = JsonBField()
+    next_version = models.OneToOneField('self', related_name='previous_version', null=True, editable=False)
```
Comment (kshepard) [inline app/ashlar/models.py]: This line's getting a bit long. Do we not have a pylintrc/setup.cfg yet?

File: app/ashlar/serializer_fields.py [hunk 2]
```diff
@@ -34,7 +36,7 @@ class JsonSchemaField(JsonBField):
     def to_internal_value(self, value):
         super(JsonSchemaField, self).to_internal_value(value)
         try:
-            jsonschema.Draft4Validator.check_schema(value)
+            SchemaModel.validate_schema(value)
```
Comment (CloudNiner) [inline app/ashlar/serializer_fields.py]: Hmmm, I actually dislike this as it adds a dependency on ashlar. Ideally I'd like to break these field/serializer classes into a separate library. Is there a sensible way to switch the dependency around?

**Comment by CloudNiner** [comment]
LGTM

File: app/ashlar/serializer_fields.py (hunk 2 above)
Comment (ddohler) [inline app/ashlar/serializer_fields.py]: Oh oops that is a really good point. Yeah I'll think of something else.

File: app/ashlar/models.py (hunk 1 above)
Comment (ddohler) [inline app/ashlar/models.py]: Good catch, I think I may have quit my editor without giving the linter a chance to work.

**Comment by kshepard** [comment]
+1

**Comment by ddohler** [comment]
@CloudNiner Broke the JSON Schema validation out into a separate validator function.

**Comment by CloudNiner** [comment]
Cool. Think this'll still work, since it can be pulled out as well in case the jsonfield stuff ends up in a different library.

**Pull request merged**